tests/test_compression.py

```diff
@@ -0,0 +1,45 @@
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from datasetinsights.io.compression import (
+    GZipCompression,
+    TarFileCompression,
+    ZipFileCompression,
+    compression_factory,
+)
+
+
+@patch("datasetinsights.io.compression._get_file_extension_from_filepath")
```

Comment:
Why are you mocking out the private function?

Reply:
Because there is no actual file to get the extension. Is there any other way?

Reply:
Also maybe there is no need to make that method private.

Reply:
I'll push some tests and we can take a look if they make sense